obspy/io/gcf/libgcf.py

```diff
@@ -0,0 +1,182 @@
+        if isLeapYear(year):
+            daycount[1] = 29
+        else:
+            daycount[1] = 28
+    return "%4d-%02d-%02d" % (year, month + 1, days + 1)
+
+
+def decodeDateTime(data):
+    '''Decode date and time field.
+    The date code is a 32 bit value specifying the start time of the block.
+    Bits 0-16 contain the number of seconds since midnight,
+    and bits 17-31 the number of days since 17th November 1989.'''
+    DATE = decodeDate(data >> 17)
+    TIME = decodeTime(data - (data >> 17 << 17))
+    starttime = UTCDateTime(DATE + 'T' + TIME)
+    return starttime
+
+
+def readDataBlock(f, skipData=False):
+    '''Read one data block from GCF file.
+    more details can be found here:
+    http://geophysics.eas.gatech.edu/GTEQ/Scream4.4/GCF_Specification.htm
+    f - file object to read from
+    if skipData is True, Only header is returned.
+    if not a data block (SPS=0) - returns None.
+    '''
+    # get ID
+    ID = f.read(4)
+    if ID == '':
+        raise EOFError  # got to EOF
+    ID = np.frombuffer(ID, count=1, dtype=np.uint32).byteswap()
+    if ID >> 31 & 0b1 > 0:
+        ID = (ID << 6) >> 6
+    ID = decode36(ID)
+    # get Stream ID
+    stID = f.read(4)
+    stID = np.frombuffer(stID, count=1, dtype=np.uint32).byteswap()
+    stID = decode36(stID)
+    # get Date & Time
+    data = f.read(4)
+    data = np.frombuffer(data, count=1, dtype=np.uint32).byteswap()
+    DATE = decodeDate(data >> 17)
+    TIME = decodeTime(data - (data >> 17 << 17))
+    starttime = UTCDateTime(DATE+'T'+TIME)
+    # get data format
+    data = f.read(4)
+    # get reserved, SPS, data type compression, number of 32bit records (N)
+    reserved, SPS, compress, N = np.frombuffer(data, count=4, dtype=np.uint8)
+    compression = compress & 0b00001111  # get compression code
+    Toffset = compress >> 4  # get time offset
+    if Toffset > 0:
+        starttime = starttime + Toffset / ToffsetsD[SPS]
+    if SPS in SPSD:
+        SPS = SPSD[SPS]  # get special SPS value if needed
+    if not SPS:
+        f.seek(N*4, 1)  # skip if not a data block
+        if 1008-N*4 > 0:
+            f.seek(1008-N*4, 1)  # keep skipping to get 1008 record
+        return None
+    NoS = int(N)*compression  # number of samples
+    header = {}
+    header['starttime'] = starttime
+    header['station'] = stID[:4]
+    header['channel'] = 'HH'+stID[4].capitalize()
+    header['sampling_rate'] = SPS
+    if skipData:
+        f.seek(4*(N+2), 1)  # skip data part (inc. FIC and RIC)
+        return header
+    else:
+        # get FIC
+        FIC = np.frombuffer(f.read(4), count=1, dtype='>i4')
+        # get incremental data
+        data = np.frombuffer(f.read(4*N), count=NoS,
+                             dtype=compressionD[compression])
+        # construct time series
+        data = (FIC + np.cumsum(data)).astype('>i4')
+        # get RIC
+        RIC = np.frombuffer(f.read(4), count=1, dtype='>i4')
+        # skip if not a full record
```

Comment:
I don't think so. Sometimes data is not filling the whole packet size and that is OK.